rust/runtime/src/module/dylib.rs

```diff
@@ -0,0 +1,90 @@
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    os::raw::{c_int, c_void},
+};
+
+use tvm_common::{ffi::BackendPackedCFunc, packed_func::PackedFunc};
+
+use crate::{
+    threading::{TVMBackendParallelBarrier, TVMBackendParallelLaunch},
+    workspace::{TVMBackendAllocWorkspace, TVMBackendFreeWorkspace},
+    TVMAPISetLastError,
+};
+
+use super::Module;
+
+pub struct DylibModule<'a> {
+    lib: libloading::Library,
+    packed_funcs: RefCell<HashMap<String, &'a (dyn PackedFunc)>>,
+}
+
+macro_rules! init_context_func {
+    ($lib:ident, $( ($fn:ident, $sig:ty) ),+ $(,)?) => {
+        unsafe {
+            $(
+                let fn_ptr = $lib.get::<*mut $sig>(concat!("__", stringify!($fn)).as_bytes());
```

Comment:
Interesting! what `__` is for? [mangling](https://docs.rs/libloading/0.5.0/libloading/struct.Library.html#method.get)?

Reply:
nope that's how the module knows where to get its context funcs (@see https://github.com/dmlc/tvm/blob/master/src/runtime/module_util.h#L43)